backend/tests/test_api/test_robots_api.py

```diff
@@ -15,6 +19,84 @@ def test_get_robots(test_app: FastAPI):
         assert len(response.json()) == 2
 
 
+def test_create_robot_success(test_app: FastAPI, session: Session):
+    body: dict = {
+        "id": 0,
+        "name": "test_robot",
```

Comment:
Comment here since it is most obvious here. I think it is a bit weird that we give an id in the POST /robots request when this is not used in the database and thus is not valid. I think we should consider different robot models for the POST /robots endpoint and the GET /robots... endpoints to better represent what information that is actually used. I also think it is unnecessary to publish host and port in the GET /robots endpoints, which would require different models for POST /robots anyway.

Reply:
I agree! I have seen this before but forgot to look into it.

Reply:
I agree! I have seen this before but forgot to look into it.

As for publishing the host and port through the GET endpoints I don't really see why not? It is static metadata for the robot that is associated with the object. It is not required for the frontend to know but I don't think we should design the data part of the API to just serve the specific frontend we have now.

Reply:
If we wish to create a different model for GET and POST we need to make adjustments to the openapi specification too.

What do you think about accepting this for now and setting this change as an issue for the workshop next week? It would be nice to have closed this PR before the workshop to avoid merge conflicts etc. as it changes quite a lot of things.

Reply:
Thats fine

Reply:
#149